apps/mobile/src/app/(tabs)/(index)/index.tsx

```diff
@@ -1,77 +1,11 @@
-import { useCollectiblesFlag } from '@/features/feature-flags';
-import { NotificationsSheet } from '@/features/notifications/notifications-sheet';
-import { useOnDetectNoNotificationPreference } from '@/features/notifications/use-notifications';
-import { RefreshControl } from '@/features/refresh-control/refresh-control';
-import { NetworkBadge } from '@/features/settings/network-badge';
-import { useTotalActivity } from '@/queries/activity/account-activity.query';
-import { useTotalCollectibles } from '@/queries/collectibles/account-collectibles.query';
-import { useWallets } from '@/store/wallets/wallets.read';
-import { t } from '@lingui/core/macro';
-import { router } from 'expo-router';
-
-import { Box, LeatherLogomarkIcon, SheetInstance } from '@leather.io/ui/native';
+import { HomeScreenWithAccount } from '@/components/home/home-with-account-screen';
+import { HomeScreenWithoutAccount } from '@/components/home/home-without-account-screen';
+import { CurrentAccountLoader } from '@/core/current-account-provider';
 
 export default function HomeScreen() {
-  const { hasWallets } = useWallets();
-  const notificationSheetRef = useRef<SheetInstance>(null);
-  const activity = useTotalActivity();
-  const collectibles = useTotalCollectibles();
-  const releaseCollectibles = useCollectiblesFlag();
-  useOnDetectNoNotificationPreference(notificationSheetRef.current?.present);
-
   return (
-    <Screen>
-      <Screen.Header
-        leftElement={
-          <Box flexDirection="row" alignItems="center" p="2" gap="2">
-            <LeatherLogomarkIcon />
-            <NetworkBadge />
-          </Box>
-        }
-        rightElement={<HeaderActions />}
-      />
-      <Screen.ScrollView refreshControl={<RefreshControl />}>
-        <Box gap="8" pt="6">
-          <AccountsWidget />
-          {hasWallets && (
-            <>
-              <BalancesWidget
-                onPressHeader={() => router.navigate('/balances')}
-                balance={<TotalBalance color="ink.text-subdued" />}
-                title={t`All tokens`}
-              >
-                <AllAccountBalancesWidget />
-              </BalancesWidget>
-              <ActivityWidget
-                activity={activity}
-                onPressHeader={() => router.navigate('/activity')}
-                title={t`All Activity`}
-              />
-              <EarnWidget />
-              {releaseCollectibles && hasCollectibles(collectibles) && (
-                <CollectiblesWidget
-                  onPressHeader={() => router.navigate('/collectibles')}
-                  title={t`All collectibles`}
-                >
-                  <Collectibles collectibles={collectibles} mode="widget" />
-                </CollectiblesWidget>
-              )}
-            </>
-          )}
-        </Box>
-      </Screen.ScrollView>
-      <NotificationsSheet sheetRef={notificationSheetRef} />
-    </Screen>
+    <CurrentAccountLoader fallback={<HomeScreenWithoutAccount />}>
```

Comment:
This works well but we should think of a better way / naming. This case of  `HomeScreenWithoutAccount` is only really when there is a new / empty wallet and quite a basic screen. 

I think a combined home screen with an early return if `!account` would work as well.